### ISSUE TITLE
Bug: llama_set_embeddings_only(true) does not actually skip lm_head computation — graph never re-reserved

### ISSUE THREAD

**Issue by Dereczoolander** [ggml-org/llama.cpp#21169]
Calling `llama_set_embeddings_only(ctx, true)` after context creation does not skip the lm_head matrix multiply. The flag is set, but the computation graph is never rebuilt, so the lm_head continues to run on every decode call. Embeddings are still returned correctly, so the bug is invisible — it just wastes compute.

This affects any use case that extracts embeddings from a generative model (RAG pipelines, semantic search, sentence embeddings, etc.).

## Root Cause

Two issues:

### 1. `set_embeddings_only()` doesn't trigger graph re-reserve

In `src/llama-context.cpp`, `set_embeddings_only()` sets the flag but does not set `sched_need_reserve = true`. Compare with `set_causal_attn()` in the same file, which correctly triggers a re-reserve:

```cpp
// set_causal_attn — correctly triggers re-reserve
void llama_context::set_causal_attn(bool value) {
    cparams.causal_attn = value;
    sched_need_reserve = true;  // ← present
}

// set_embeddings_only — missing re-reserve trigger
void llama_context::set_embeddings_only(bool value) {
    cparams.embeddings_only = value;
    if (value) {
        cparams.embeddings = true;
    }
    // sched_need_reserve = true is NOT set
}
```

Without `sched_need_reserve = true`, the computation graph is reused as-is from the previous reserve, which includes the lm_head.

### 2. Multiple model builders missing `embeddings_only` guard

Even with the re-reserve fix, the graph still includes the lm_head for most architectures. The model graph builders unconditionally include `build_lora_mm(model.output, cur)` (the lm_head). Only `src/models/llama.cpp` has a guard:

```cpp
// llama.cpp — has the guard
if (!cparams.embeddings_only) {
    cur = build_lora_mm(model.output, cur);
    ...
    res->t_logits = cur;
}
```

The following architectures are **missing** this guard:

- `src/models/qwen35.cpp`
- `src/models/qwen3moe.cpp`
- `src/models/qwen35moe.cpp`
- `src/models/gemma2-iswa.cpp`
- `src/models/seed-oss.cpp`

(There are likely more — these are just the ones I tested. Any model builder that doesn't check `cparams.embeddings_only` before the lm_head has this bug.)

## How to Reproduce

```cpp
// Create context normally
llama_context * ctx = llama_init_from_model(model, ctx_params);

// Enable embeddings_only (should skip lm_head)
llama_set_embeddings_only(ctx, true);

// Decode — lm_head still runs
llama_decode(ctx, batch);

// Embeddings are correct, but lm_head compute was wasted
```

You can verify by checking the graph node count — it stays the same before and after `set_embeddings_only(true)`. With the fix, the node count drops (e.g., 1833 → 1832 for Qwen 3.5 9B).

## Impact

Measured on Qwen 3.5 9B (Q8_0) with RTX 3090:

| Config | Wasted lm_head compute per token | % overhead |
|--------|----------------------------------|-----------|
| GPU (ngl=99) | 1.52 ms | 10.3% |
| CPU (ngl=0) | 46.30 ms | 12.9% |

Cross-model measurements show the waste ranges from 1.5% (LLaMA-7B, 32K vocab) to 29% (Qwen3.5-0.8B, 248K vocab) depending on vocabulary size and model architecture.

### Fix 1: `src/llama-context.cpp`

```cpp
void llama_context::set_embeddings_only(bool value) {
    if (cparams.embeddings_only == value) {
        return;
    }
    cparams.embeddings_only = value;
    if (value) {
        cparams.embeddings = true;
    }
    sched_need_reserve = true;  // <-- ADD THIS
}
```

### Fix 2: All model builders

Add `if (!cparams.embeddings_only)` guard around lm_head in every model's `build_graph`:

```cpp
res->t_embd = cur;

if (!cparams.embeddings_only) {
    cur = build_lora_mm(model.output, cur);
    cb(cur, "result_output", -1);
    res->t_logits = cur;
}
```

Ideally this check should be enforced in the base class or a shared helper so new model implementations can't accidentally omit it.

## Environment

- llama.cpp commit: `2405d59` (HEAD as of testing)
- GPU: NVIDIA RTX 3090
- CUDA: 12.0
- OS: Ubuntu 22.04

## Verified Architectures

| Architecture | File                                                      | Had guard? |
|--------------------|-------------------------------------------------|-----------------|
| llama             | `src/models/llama.cpp`                   | Yes (already correct) |
| qwen35         | `src/models/qwen35.cpp`              | **No** |
| qwen3moe   | `src/models/qwen3moe.cpp`        | **No** |
| qwen35moe | `src/models/qwen35moe.cpp`      | **No** |
| gemma2       | `src/models/gemma2-iswa.cpp`   | **No** |
| seed_oss       | `src/models/seed-oss.cpp`             | **No** |
